arcgis-java-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/UtilityNetworkTraceTool.java

```diff
@@ -0,0 +1,999 @@
+          removedResult.selectResultFeatures(false);
+          mapView.getGraphicsOverlays().remove(removedResult.getResultsGraphicsOverlay());
+        }
+      }
+    });
+
+    // listener for the starting points symbol property
+    startingPointSymbolProperty.addListener((observable, oldValue, newValue) -> {
+      // when the property changes, update the existing starting points' symbology
+      startingPointsProperty.get().forEach(startingPoint -> startingPoint.updateSelectionGraphicSymbol(newValue));
+    });
+
+    // enable trace is true if there is a trace configuration selected and sufficient starting points
+    enableTraceProperty.bind(
+      Bindings.and(selectedTraceConfigurationProperty.isNotNull(), insufficientStartingPointsProperty.not()));
+
+    // load and configure the data
+    setupUtilityNetworks();
+
+    getStyleClass().add(DEFAULT_STYLE_CLASS);
+
+    setMinHeight(USE_PREF_SIZE);
+    setMinWidth(USE_PREF_SIZE);
+    setMaxHeight(USE_PREF_SIZE);
+    setMaxWidth(USE_PREF_SIZE);
+  }
+
+  @Override
+  public String getUserAgentStylesheet() {
+    return Objects.requireNonNull(this.getClass().getResource("skins/utility-network-trace.css")).toExternalForm();
+  }
+
+  /**
+   * Returns a read-only property containing the MapView this UtilityNetworkTrace is linked to.
+   *
+   * @return the MapView Property as read-only
+   * @since 100.15.0
+   */
+  public ReadOnlyObjectProperty<MapView> mapViewReadOnlyProperty() { return mapViewProperty.getReadOnlyProperty(); }
+
+  /**
+   * Gets the MapView this UtilityNetworkTrace is linked to.
+   *
+   * @return the map view
+   * @since 100.15.0
+   */
+  public MapView getMapView() {
+    return mapViewProperty.get();
+  }
+
+  /**
+   * Returns a read-only property containing the selected Utility Network so that changes to the selected Utility
+   * Network can be observed.
+   *
+   * @return the selectedUtilityNetwork Property as read-only
+   * @since 100.15.0
+   */
+  public ReadOnlyObjectProperty<UtilityNetwork> selectedUtilityNetworkReadOnlyProperty() {
+    return selectedUtilityNetworkProperty.getReadOnlyProperty();
+  }
+
+  /**
+   * Returns a read-only property containing the last completed trace as a {@link UtilityNetworkTraceToolCompletedTrace}
+   * so that completed traces can be observed and containing data accessed.
+   *
+   * @return a read-only property for the {@link UtilityNetworkTraceToolCompletedTrace}
+   * @since 100.15.0
+   */
+  public ReadOnlyObjectProperty<UtilityNetworkTraceToolCompletedTrace> completedTraceReadOnlyProperty() {
+    return completedTraceProperty.getReadOnlyProperty();
+  }
+
+  /**
+   * Property that determines whether a MapView's onMouseClicked event is used for adding starting points.
+   * Defaults to true.
+   *
+   * @return the isAddingStartingPoints property
+   * @since 100.15.0
+   */
+  public SimpleBooleanProperty isAddingStartingPointsProperty() { return isAddingStartingPointsProperty; }
```

Comment:
After looking at the design it seems that this is not well named since it looks like even if this is false I can still add starting points through other means. I think the name should reflect that this refers to disabling a user interaction.

Reply:
I've fed this back but a change won't make it in in this iteration. In the meantime I've added some additional doc on the add starting point methods to note this.